inSociety/Auth & Sign in/AuthViewController.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  ViewController.swift
+//  inSociety
+//
+//  Created by Aleksey Libin on 12.10.2022.
+//
+
+import UIKit
+import FirebaseAuth
+import FirebaseCore
+import GoogleSignIn
+
+class AuthViewController: UIViewController {
+    
+    private let logoImage = UIImageView(named: "inSociety", contentMode: .scaleAspectFit)
+    private let googleLabel = UILabel(text: "Get started with")
+    private let emailLabel = UILabel(text: "Or sign up with")
+    private let loginLabel = UILabel(text: "Already on board?")
+    
+    private let googleButton = UIButton(title: "Google", titleColor: .black, backgroundColor: .white)
+    private let emailButton = UIButton(title: "email", titleColor: .black, backgroundColor: .mainYellow())
+    private let loginButton = UIButton(title: "Login", titleColor: .mainYellow(), backgroundColor: nil)
+    
+    private let signUpVC = SignUpViewController()
+    private let loginVC = LoginViewController()
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setupViews()
+        
+        emailButton.addTarget(self, action: #selector(emailButtonPressed), for: .touchUpInside)
+        loginButton.addTarget(self, action: #selector(loginButtonPressed), for: .touchUpInside)
+        googleButton.addTarget(self, action: #selector(googleButtonPressed), for: .touchUpInside)
+        
+        signUpVC.delegate = self
+        loginVC.delegate = self
+        
+    }
+    
+}
+
+
+
+//MARK: - Buttons realization
+extension AuthViewController {
+    
+    @objc private func emailButtonPressed() {
+        present(signUpVC, animated: true)
+    }
+    
+    @objc private func loginButtonPressed() {
+        present(loginVC, animated: true)
+    }
+    
+    @objc private func googleButtonPressed() {
+        let clientID = FirebaseApp.app()?.options.clientID
+        AuthService.shared.googleLogin(clientID: clientID, presentingVC: self) { result in
+            switch result {
+            case .success(let user):
+                FirestoreService.shared.getUserData(user: user) { result in
+                    switch result {
+                    case .success(let userModel):
+                        
+                        let main = MainTabBarController(currentUser: userModel)
+                        main.modalPresentationStyle = .fullScreen
+                        self.present(main, animated: true)
+                    case .failure(_):
+                        self.showAlert(with: "You have successfully registrated") {
+                            self.present(SetupProfileViewController(currentUser: user), animated: true)
+                        }
+                    }
+                }
+            case .failure(let failure):
+                self.showAlert(with: "Error", and: failure.localizedDescription)
+            }
+        }
+    }
+}
+
+
+
+//MARK: - Setup views
+extension AuthViewController {
```

Comment:
the same as above and no needed to have a lot of extensions, you can connect all setup methods in one extension